### PR TITLE
Prepare MongooseIM 4.2.0 release

### PR DESCRIPTION
**Pull request by leszke**
It's happening! 🕺 

TODO:
- [x] to update Migration Guide with `archive_groupchats` info - @chrzaszcz 
- [x] to add anything we would like to have included in this release, that will get merged to master branch in-between today and release date
- [x] to add `4.2.0` milestone - @leszke 
- [x] to add `4.2.0` tag - @leszke (tag will be created when we publish)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,53 @@
...
+- Usage of deprecated `http_uri` in `ejabberd_auth_http` and `mod_muc_room` modules (#3026)
+- Config reload functionality with the flat config format (#3030)
+
+### Changed
+- Usage of maps instead of lists in session management (#3018)
+- Improved timestamp logic (#3031)
+- Prepared queries for MS SQL, mySQL and pgSQL (#3039, #3050, #3055, #3059, #3060, #3066, #3074, #3078)
+- Improved pipeline for `mod_smart_markers` (#3068)
+- Unified checks for chat markers (#3080)
+- Set `mod_mam_meta.pm.archive_groupchats` to `false` by default (#3082)
```
Comment (chrzaszcz) [inline CHANGELOG.md:26]: Hm... maybe this should be "fixed"? According to the docs they should've been set to 'false' already.

**Review by chrzaszcz — COMMENTED** [review]
I reviewed the changelog with a few comments. Could we bring back the `## Special thanks to our contributors:` section? It was there for the previous releases and we have a few contributors form the community. Also, @NelsonVides what do you think about including Beekeeper there?

**Review by chrzaszcz — APPROVED** [review]
Great!

**Review by NelsonVides — APPROVED** [review]
Love it! 🔥

**Pull request merged**